include/aspect/particle/property_pool.h

```diff
@@ -0,0 +1,100 @@
+/*
+ Copyright (C) 2016 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __aspect__particle_property_pool_h
+#define __aspect__particle_property_pool_h
+
+#include <aspect/global.h>
+
+//#include <aspect/particle/particle.h>
+#include <deal.II/base/array_view.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    using namespace dealii;
+
+    /**
+     * This class manages the memory space in which particles store their
+     * properties. Because this is dynamic memory and every particle needs the
+     * same amount it is more efficient to let this be handled by a central
+     * manager that does not need to allocate/deallocate memory every time a
+     * particle is constructed/destroyed.
+     */
+    class PropertyPool {
+      public:
+        struct Handle {
+          public:
+          Handle()
+          :
+            index(numbers::invalid_unsigned_int)
+          {}
+
+          Handle(const std::size_t index)
```

Comment:
I think it would be good to use a typedef in class `PropertyPool` instead of a hard-coded `std::size_t`. We may want to use a 32-bit integer, for example, assuming that no processor stores more than 4B particles.